plugins/sentinelone/komand_sentinelone/tasks/monitor_activities_and_events/task.py

```diff
@@ -0,0 +1,327 @@
+                else:
+                    self.logger.info(f"Pagination in progress, skipping {log_type} collection...")
+        return all_logs, total_forbidden_responses
+
+    def get_generic_logs(
+        self,
+        log_type: str,
+        state: dict,
+        total_forbidden_responses: int,
+        current_run_timestamp: str,
+        last_run_timestamp: str = None,
+        pagination_cursor: str = None,
+    ) -> List[Dict]:
+        """
+        Determine log type and query that log set with query parameters determined by type, cursor, and last timestamp.
+        Set latest log timestamp and page cursor in state.
+        Return logs from query
+        """
+        timestamp_key = LOG_TYPE_MAP.get(log_type, {}).get("LAST_LOG_TIMESTAMP")
+        page_token_key = LOG_TYPE_MAP.get(log_type, {}).get("PAGE_TOKEN")
+        last_log_timestamp = state.get(timestamp_key)
+        if not last_log_timestamp:
+            last_log_timestamp = STARTING_TIMESTAMP
+        query_params = self.get_query_params(log_type, last_run_timestamp, current_run_timestamp, pagination_cursor)
+        if log_type == ACTIVITIES_LOGS:
+            response = self.connection.client.get_activities_list(query_params, True, False)
+        elif log_type == EVENTS_LOGS:
+            response = self.connection.client.get_device_control_events(query_params, True, False)
+        else:
+            response = self.connection.client.get_threats(query_params, API_VERSION, True, False)
+        try:
+            status_code = response.status_code
+            if response.status_code in [401, 403]:
+                self.logger.info(f"Received status code {status_code} for {log_type} query")
+            response_handler(response, allowed_status_codes=[401, 403])
+        except PluginException as error:
+            raise ApiException(cause=error.cause, assistance=error.assistance, status_code=status_code, data=response)
+        logs, next_page_cursor, total_forbidden_responses = self.extract_query_response(
+            response, total_forbidden_responses
+        )
+        new_logs_last_timestamp = self.get_latest_timestamp(logs, last_log_timestamp)
+        state[timestamp_key] = new_logs_last_timestamp
+        state[page_token_key] = next_page_cursor
+        if next_page_cursor:
+            self.logger.info(f"New next page cursor received for {log_type}: {next_page_cursor}")
+        return logs, total_forbidden_responses
+
+    def get_query_params(self, log_type: str, last_log_timestamp: str, current_run_timestamp: str, cursor: str) -> Dict:
+        """
+        Generate query parameters based on log type and availability of pagination cursor
+        """
+        timestamp_name = "createdAt"
+        if log_type == EVENTS_LOGS:
+            timestamp_name = "eventTime"
+        query_params = {
+            "limit": LOGS_PAGE_LIMIT,
+            f"{timestamp_name}__gt": last_log_timestamp,
+            f"{timestamp_name}__lte": current_run_timestamp,
```

Comment:
do these params have no impact on the results if the cursor is supplied? should we only add them into `query_params` if there is no cursor?

Reply:
They do have an impact if changed and can modify the results provided. They shouldn't change, however best not to take any chances.